shark_turbine/kernel/wave/wave.py

```diff
@@ -170,6 +172,10 @@ def _trace_and_get_kernel_signature(
         idxc = IndexingContext.current()
         idxc.finalize()
 
+        # Promote the placeholders to the appropriate address space.
+        promote_placeholders(graph)
+        hoist_allocs(graph)
```

Comment:
Can we add a simple codegen test for this?

Reply:
That was my stretch goal for this week :) Will add it after I upload a minimize global load PR.

Reply:
haha sweet, will leave this for a todo/follow on PR then.